eng/scripts/automation/GenerateAndBuildLib.ps1

```diff
@@ -284,65 +331,299 @@ function Invoke-Generate() {
+    Write-Host "readmeFile:$readmeFile"
+    $service, $serviceType = Get-ResourceProviderFromReadme $readmeFile
+    Write-Host "service:$service, serviceType:$serviceType"
+    
+    if (!$readmeFile.StartsWith("http") -And !(Test-Path -Path $readmeFile)) {
+        Write-Error "readme file does not exist."
+        exit 1
+    }
+    
+    $packagesToGen = @()
+    $newpackageoutput = "newPackageOutput.json"
+    if ( $serviceType -eq "resource-manager" ) {
+        Write-Host "Generate resource-manager SDK client library."
+        $package = $service
+        if ($packageNameHash[$service] -ne "") {
+            $package = $packageNameHash[$service]
         }
-        if ($readmeFile -match $readmeFileRegex)
-        {
-            return $matches["specName"], $matches["serviceType"]
+        New-MgmtPackageFolder -service $service -packageName $package -sdkPath $sdkRootPath -commitid $commitid -readme $readmeFile -outputJsonFile $newpackageoutput
+        if ( !$?) {
+            Write-Error "Failed to create sdk project folder. exit code: $?"
+            exit 1
+        }
+        $newpackageoutputJson = Get-Content $newpackageoutput | Out-String | ConvertFrom-Json
+        $packagesToGen = $packagesToGen + @($newpackageoutputJson)
+        Remove-Item $newpackageoutput
+    } else {
+        Write-Host "Generate data-plane SDK client library."
+        $namespace = ""
+        if ( $autorestConfigYaml) {
+            # support single package
+            $range = ($autorestConfigYaml | Select-String -Pattern '```').LineNumber
+            if ( $range.count -gt 1) {
+                $startNum = $range[0];
+                $lines = $range[1] - $range[0] - 1
+                $autorestConfigYaml = ($autorestConfigYaml | Select -Skip $startNum | Select -First $lines) |Out-String
+            }
+            Install-Module -Name powershell-yaml -Force -Verbose -Scope CurrentUser
+            Import-Module powershell-yaml
+            $yml = ConvertFrom-YAML $autorestConfigYaml
+
+            $outputFolder = $yml["output-folder"]
+            if ($outputFolder) {
+                $directories = $outputFolder.Split("/");
+                $service = $directories[-2];
+                $namespace = $directories[-1];
+            }
+
+            New-DataPlanePackageFolder -service $service -namespace $namespace -sdkPath $sdkRootPath -readme $readmeFile -autorestConfigYaml "$autorestConfigYaml" -outputJsonFile $newpackageoutput
+            if ( !$? ) {
+                Write-Error "Failed to create sdk project folder. exit code: $?"
+                exit 1
+            }
+            $newpackageoutputJson = Get-Content $newpackageoutput | Out-String | ConvertFrom-Json
+            $packagesToGen = $packagesToGen + @($newpackageoutputJson)
+            Remove-Item $newpackageoutput
+        } else {
+            npx autorest --version=3.8.4 --csharp $readmeFile --csharp-sdks-folder=$sdkRootPath --skip-csproj --clear-output-folder=true
```

Comment:
Given our new plans I don't think we will ever be in this else block. If we don't have an autorest config we will either create one if we have the data from the input or we will skip for this library.

Reply:
This part is try to support existing rare scenarios which may have multiple sdk packages in one RP and it define batch task in the readme.csharp.md.

Reply:
I would like to avoid having this case if possible. Can you point me at an example in the specs repo that would hit this case?

Reply:
Hi @weshaggard Purview is such kind of RP, in sdk\purview it will contain more than one sdk packages (administrator, catalog, scanning), and there will be only one readme file. In order to support this, we need to use `batch` to defined in readme.md to support generate all those packages at once.

Reply:
@dw511214992 @josefree what is our strategy for handling batches? I know we had ah email about how that data would be passed for this multiple sdk case.